compose/service.py

```diff
@@ -936,20 +937,55 @@ def build_volume_binding(volume_spec):
     return volume_spec.internal, "{}:{}:{}".format(*volume_spec)
 
 
+def ensure_engine_compatibility(external_path, internal_path):
```

Comment:
That naming, was me trying to describe "what" we're doing and part of that what included the "why". When naming things, especially tricky things like this, I try to think about the person who has to come after me and fix it or modify the code, so I lean towards verbosity and whys as that will aid them to understand why the code is the way it is and pick it up quicker than it took me to figure it out the first time.

Ultimately we're agreeing with each other but with different words :)

I'm happier with `normalize_paths_for_engine` for now, as this is a workaround a hack within the engine itself. Then when that is no longer the case, we can rename it again. Cool.